client/ayon_usd/config.py

```diff
@@ -0,0 +1,175 @@
+    "file_logger_file_path",
+)
+
+ADDON_SETTINGS_USD_RESOLVER_LOG_LOGGIN_KEYS = (
+    "Ayon_UsdResolver_Settings",
+    "ayon_logger_logging_keys",
+)
+
+
+def get_addon_settings_value(settings: dict, key_path: tuple):
+    try:
+        selected_element = settings
+        for key in key_path:
+            selected_element = selected_element[key]
+
+        return selected_element
+    except (KeyError, TypeError) as e:
+        raise KeyError(f"Error accessing settings with key path {key_path}: {e}")
+
+
+class SingletonFuncCache:
+    _instance = None
+    _cache = {}
+
+    def __new__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    @classmethod
+    def cache(cls, func):
+        @functools.wraps(func)
+        def cache_func(*args, **kwargs):
+
+            cache_key = tuple((func.__name__, cls._hash_args_kwargs(args, kwargs)))
+
+            if cache_key in cls._cache.keys():
+                return cls._cache[cache_key]
+            result = func(*args, **kwargs)
+            cls._cache[cache_key] = result
+
+            return result
+
+        return cache_func
+
+    @staticmethod
+    def _hash_args_kwargs(args, kwargs):
+        """Generate a hashable key from *args and **kwargs."""
+        args_hash = SingletonFuncCache._make_hashable(args)
+        kwargs_hash = SingletonFuncCache._make_hashable(kwargs)
+        return args_hash + kwargs_hash
+
+    @staticmethod
+    def _make_hashable(obj):
+        """Converts an object to a hashable representation."""
+
+        if isinstance(obj, (int, float, str, bool, type(None))):
+            return hashlib.sha256(str(obj).encode()).hexdigest()
+
+        if isinstance(obj, dict) or hasattr(obj, "__dict__"):
+            return hashlib.sha256(json.dumps(obj, sort_keys=True).encode()).hexdigest()
+
+        try:
+            return hashlib.sha256(json.dumps(obj).encode()).hexdigest()
+        except TypeError:
+            return hashlib.sha256(str(id(obj)).encode()).hexdigest()
+
+    def debug(self):
+        return self._cache
+
+
+def print_cache():
+    print(SingletonFuncCache().debug())
+
+
+def get_addon_settings():
+
+    return ayon_api.get_addon_settings(
+        addon_name=ADDON_NAME,
+        addon_version=ADDON_VERSION,
+        variant=AYON_BUNDLE_NAME,
+    )
+
+
+@SingletonFuncCache.cache
+def get_global_lake_instance():
+    addon_settings = (
+        get_addon_settings()
+    )  # the function is cached, but this reduces the call stack
+    return wrapper.LakeCtl(
+        server_url=str(
+            get_addon_settings_value(addon_settings, ADDON_SETTINGS_LAKE_FS_URI)
+        ),
+        access_key_id=str(
+            get_addon_settings_value(addon_settings, ADDON_SETTINGS_LAKE_FS_KEY_ID)
+        ),
+        secret_access_key=str(
+            get_addon_settings_value(addon_settings, ADDON_SETTINGS_LAKE_FS_KEY)
+        ),
+    )
```

Comment:
Why are you using those constants? It make the code really hard to read and find out what is actually used...
```suggestion
    # the function is cached, but this reduces the call stack
    addon_settings = get_addon_settings()
    lake_fs_settings = addon_settings["LakeFs_Settings"]
    return wrapper.LakeCtl(
        server_url=lake_fs_settings["ayon_usd_lake_fs_server_uri"]
        access_key_id=lake_fs_settings["access_key_id"],
        secret_access_key=lake_fs_settings["secret_access_key"],
    )
```

Reply:
It uses constants for a few reasons. 
1. We can refactor the settings layout and names without having to refactor more than one place in the add-on. 
2. it makes the code easy to read and understand because knowing what keys you need in a dict is unnecessary. If you want the LakeFs uri, you take the Settings constant for the LakeFs Uri. 
3. it makes searching the code base easy. 
4. it makes the code simple to read as dict access syntax is eliminated and replaced with a simple Var name that thoroughly explains itself.